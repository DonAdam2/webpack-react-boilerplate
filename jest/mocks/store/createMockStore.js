import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middleWares = [thunk];

export default (initialState) => {
  // create a module for creating different states of the store
  const mockStore = configureMockStore(middleWares)(initialState);

  // replace the dispatch method with a spy and keep the functionality
  mockStore.dispatch = jest.fn(mockStore.dispatch);
  return mockStore;
};
