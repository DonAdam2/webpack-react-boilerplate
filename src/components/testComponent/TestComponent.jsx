import { useDispatch, useSelector } from 'react-redux';

import { getTestAction } from '@/store/app/selectors/AppSelectors';
import { updateTestString } from '@/store/app/slices/AppSlice';

const TestComponent = () => {
  const dispatch = useDispatch(),
    testAction = useSelector((state) => getTestAction({ state }));

  return (
    <div className="container">
      <p>
        Current environment API is <strong>{process.env.BASE_URL}</strong>
      </p>
      <p>
        Testing the store <strong>{testAction}</strong>
      </p>
      <button onClick={() => dispatch(updateTestString())}>Change text</button>
    </div>
  );
};

export default TestComponent;
