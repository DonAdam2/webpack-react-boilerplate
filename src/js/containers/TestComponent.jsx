import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
//selectors
import { getTestAction } from '../store/app/selectors/AppSelectors';
//actions
import { setTestAction } from '../store/app/actions/AppActions';

const TestComponent = () => {
	const dispatch = useDispatch(),
		testAction = useSelector((state) => getTestAction({ state }));

	return (
		<div className="container">
			<p>
				Current environment API is <strong>{process.env.BASE_URL}</strong>
			</p>
			<p>
				Testing the store <strong>{testAction}</strong>
			</p>
			<button onClick={() => dispatch(setTestAction())}>Change text</button>
		</div>
	);
};

export default TestComponent;
