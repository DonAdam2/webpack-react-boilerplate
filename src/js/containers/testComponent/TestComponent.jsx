import { useDispatch, useSelector } from 'react-redux';
//selectors
import { getTestAction } from '@/js/store/app/selectors/AppSelectors';
//actions
import { updateTestString } from '@/js/store/app/slices/AppSlice';

const TestComponent = () => {
  const dispatch = useDispatch(),
    testAction = useSelector((state) => getTestAction({ state }));

  return (
    <div className="container">
      <p>
        Current environment API is <strong>{process.env.BASE_URL}</strong>
      </p>
      <p>
        Testing the store <strong>{testAction}</strong>
      </p>
      <button onClick={() => dispatch(updateTestString())}>Change text</button>
    </div>
  );
};

export default TestComponent;
