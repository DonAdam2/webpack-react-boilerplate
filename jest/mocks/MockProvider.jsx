import React from 'react';
import { Provider } from 'react-redux';
// default store
import store from './store';

const MockProvider = ({ children, mockStore = store }) => (
  <Provider store={mockStore}>{children}</Provider>
);

export default MockProvider;
