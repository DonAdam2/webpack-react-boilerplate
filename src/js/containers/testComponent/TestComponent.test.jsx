import React from 'react';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';
// snapshots renderer
import renderer from 'react-test-renderer';
// mock store provider
import MockProvider from '@/jest/mock/MockProvider';
// mock store
import createMockStore from '@/jest/mock/store/createMockStore';
//components
import TestComponent from './TestComponent';

afterEach(cleanup);

describe('testComponent.jsx', () => {
	it('snapshot renders correctly, truthy values', () => {
		const store = createMockStore({
			app: {
				testString: 'Initial test',
			},
		});

		const tree = renderer
			.create(
				<MockProvider mockStore={store}>
					<TestComponent />
				</MockProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should dispatch TEST_ACTION action when user clicks the button', () => {
		const store = createMockStore({
			app: {
				testString: 'Initial test',
			},
		});
		render(
			<MockProvider mockStore={store}>
				<TestComponent />
			</MockProvider>
		);
		fireEvent.click(screen.getByTestId('changeText'));
		expect(store.dispatch).toHaveBeenCalledTimes(1);
	});
});
