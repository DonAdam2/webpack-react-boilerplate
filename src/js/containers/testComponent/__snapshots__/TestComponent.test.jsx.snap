// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testComponent.jsx snapshot renders correctly, truthy values 1`] = `
<div
  className="container"
>
  <p>
    Current environment API is 
    <strong />
  </p>
  <p>
    Testing the store 
    <strong>
      Initial test
    </strong>
  </p>
  <button
    data-testid="changeText"
    onClick={[Function]}
  >
    Change text
  </button>
</div>
`;
